java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -306,28 +329,63 @@ public Builder maxInboundMessageSize(int maxMessageSize) {
       return this;
     }
 
+    /**
+     * A small utility function to ensure that InputStream attributes.
+     * are closed if they are not null
+     * @param stream The InputStream to close (if it is not null).
+     */
+    private void closeInputStreamIfNotNull(InputStream stream) throws IOException {
+      if (stream != null) {
+        stream.close();
+      }
+    }
+
     /**
      * Enable TLS on the server.
      * @param certChain The certificate chain to use.
      * @param key The private key to use.
      */
     public Builder useTls(final File certChain, final File key) throws IOException {
+      closeInputStreamIfNotNull(this.certChain);
+      closeInputStreamIfNotNull(this.key);
```

Comment:
We should set these to null after closing, because there can be an exception opening the new inputs. So this would leave these fields in a bad state (holding closed streams).

Reply:
would it be like this?:
```
    private void closeInputStreamIfNotNull(InputStream stream) throws IOException {
      if (stream != null) {
        stream.close();
      }
      stream = null;
    }
```

Reply:
No, because that only sets the local - you need to set `this.certChain` etc to null

Reply:
I believe I've addressed these suggestions.  Please confirm.  Thanks.

Reply:
Also - I'm unsure as to Pull Request protocol - should I hit the "Resolve conversation" button when I believe I've addressed the concern, or is that up to the reviewer to do?